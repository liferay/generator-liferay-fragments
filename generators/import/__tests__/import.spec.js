// @ts-nocheck

jest.mock('../../compress/compress');
jest.mock('../../../utils/api');

const api = require('../../../utils/api');
const compress = require('../../compress/compress');
const getTestFixtures = require('../../../utils/get-test-fixtures');
const { ADD_DEPLOYMENT_DESCRIPTOR_VAR } = require('../../../utils/constants');
const importProject = require('../import');
const JSZip = require('jszip');

const GROUP_ID = '1234';

getTestFixtures().forEach(projectPath => {
  describe(`import ${projectPath}`, () => {
    const zip = new JSZip();
    zip.file('sample.txt', 'sample');

    afterEach(() => {
      compress.mockReset();
      api.importZip.mockReset();
    });

    beforeEach(() => {
      compress.mockImplementation(() => Promise.resolve(zip));
      api.importZip.mockImplementation(() => Promise.resolve({}));
    });

    it('tries to generate a zip file from the given project', async () => {
      await importProject(GROUP_ID, projectPath);

      expect(compress).toHaveBeenCalledWith(projectPath, {
        [ADD_DEPLOYMENT_DESCRIPTOR_VAR]: false
      });
    });

    it('sends the given zip to backend', async () => {
      await importProject(GROUP_ID, projectPath);
      expect(api.importZip).toHaveBeenCalledWith(projectPath, GROUP_ID);
    });

    it('imports the project using old APIs if compress error', async () => {
      compress.mockImplementation(() => {
        throw new Error('error');
      });

      const legacyMock = jest.spyOn(importProject, 'legacy');
      legacyMock.mockImplementation(() => {});

      await importProject(GROUP_ID, projectPath);
      expect(legacyMock).toHaveBeenCalled();
    });

    it('imports the project using old APIs if api error', async () => {
      api.importZip.mockImplementation(() => {
        throw new Error('error');
      });

      const legacyMock = jest.spyOn(importProject, 'legacy');
      legacyMock.mockImplementation(() => {});

      await importProject(GROUP_ID, projectPath);
      expect(legacyMock).toHaveBeenCalled();
    });

    test.todo('[DEPRECATED] creates every collection in local project');
    test.todo('[DEPRECATED] updates existing collections');
    test.todo('[DEPRECATED] imports every fragment in local project');
    test.todo('[DEPRECATED] updates existing fragments');
  });
});
