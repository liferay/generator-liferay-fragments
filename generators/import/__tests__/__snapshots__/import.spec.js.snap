// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import-generator/import imports collections to a Liferay Site: addFragmentCollection 1`] = `
Array [
  "1234",
  "sample-collection",
  "Sample collection name",
  "Sample collection description",
]
`;

exports[`import-generator/import imports collections to a Liferay Site: addFragmentCollection 2`] = `
Array [
  "1234",
  "sample-collection",
  "Sample collection name",
  "Sample collection description",
]
`;

exports[`import-generator/import imports collections to a Liferay Site: addFragmentCollection 3`] = `
Array [
  "1234",
  "sample-collection",
  "Sample collection name",
  "Sample collection description",
]
`;

exports[`import-generator/import imports fragments to a Liferay site: addFragmentEntry 1`] = `
Array [
  "1234",
  "sample-collection",
  "sample-component-fragment",
  Object {
    "configuration": "{
  \\"fieldSets\\": []
}
",
    "css": ".fragment { color: inherit; }
",
    "html": "<div>Fragment</div>
",
    "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
    "name": "Sample component fragment name",
    "previewFileEntryId": 0,
    "status": 0,
    "type": 1,
  },
]
`;

exports[`import-generator/import imports fragments to a Liferay site: addFragmentEntry 2`] = `
Array [
  "1234",
  "sample-collection",
  "sample-fragment",
  Object {
    "configuration": "{
  \\"fieldSets\\": []
}
",
    "css": ".fragment { color: inherit; }
",
    "html": "<div>Fragment</div>
",
    "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
    "name": "Sample fragment name",
    "previewFileEntryId": 0,
    "status": 0,
    "type": 0,
  },
]
`;

exports[`import-generator/import imports fragments to a Liferay site: addFragmentEntry 3`] = `
Array [
  "1234",
  "sample-collection",
  "sample-section-fragment",
  Object {
    "configuration": "{
  \\"fieldSets\\": []
}
",
    "css": ".fragment { color: inherit; }
",
    "html": "<div>Fragment</div>
",
    "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
    "name": "Sample section fragment name",
    "previewFileEntryId": 0,
    "status": 0,
    "type": 0,
  },
]
`;

exports[`import-generator/import imports fragments to a Liferay site: addFragmentEntry 4`] = `
Array [
  "1234",
  "sample-collection",
  "sample-fragment",
  Object {
    "configuration": "{
  \\"fieldSets\\": []
}
",
    "css": ".fragment { color: inherit; }
",
    "html": "<div>Fragment</div>
",
    "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
    "name": "Sample fragment name",
    "previewFileEntryId": 0,
    "status": 0,
    "type": 0,
  },
]
`;

exports[`import-generator/import imports fragments to a Liferay site: addFragmentEntry 5`] = `
Array [
  "1234",
  "sample-collection",
  "sample-fragment",
  Object {
    "configuration": "",
    "css": ".fragment { color: inherit; }
",
    "html": "<div>Fragment</div>
",
    "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
    "name": "Sample fragment name",
    "previewFileEntryId": 0,
    "status": 0,
    "type": 0,
  },
]
`;

exports[`import-generator/import imports fragments to a Liferay site: updateFragmentCollection 1`] = `
Array [
  "sample-collection",
  "Sample collection name",
  "Sample collection description",
]
`;

exports[`import-generator/import imports fragments to a Liferay site: updateFragmentCollection 2`] = `
Array [
  "sample-collection",
  "Sample collection name",
  "Sample collection description",
]
`;

exports[`import-generator/import imports fragments to a Liferay site: updateFragmentCollection 3`] = `
Array [
  "sample-collection",
  "Sample collection name",
  "Sample collection description",
]
`;

exports[`import-generator/import updates collections if they already exist: addFragmentEntry 1`] = `
Array [
  "1234",
  "sample-collection",
  "sample-component-fragment",
  Object {
    "configuration": "{
  \\"fieldSets\\": []
}
",
    "css": ".fragment { color: inherit; }
",
    "html": "<div>Fragment</div>
",
    "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
    "name": "Sample component fragment name",
    "previewFileEntryId": 0,
    "status": 0,
    "type": 1,
  },
]
`;

exports[`import-generator/import updates collections if they already exist: addFragmentEntry 2`] = `
Array [
  "1234",
  "sample-collection",
  "sample-fragment",
  Object {
    "configuration": "{
  \\"fieldSets\\": []
}
",
    "css": ".fragment { color: inherit; }
",
    "html": "<div>Fragment</div>
",
    "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
    "name": "Sample fragment name",
    "previewFileEntryId": 0,
    "status": 0,
    "type": 0,
  },
]
`;

exports[`import-generator/import updates collections if they already exist: addFragmentEntry 3`] = `
Array [
  "1234",
  "sample-collection",
  "sample-section-fragment",
  Object {
    "configuration": "{
  \\"fieldSets\\": []
}
",
    "css": ".fragment { color: inherit; }
",
    "html": "<div>Fragment</div>
",
    "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
    "name": "Sample section fragment name",
    "previewFileEntryId": 0,
    "status": 0,
    "type": 0,
  },
]
`;

exports[`import-generator/import updates collections if they already exist: addFragmentEntry 4`] = `
Array [
  "1234",
  "sample-collection",
  "sample-fragment",
  Object {
    "configuration": "{
  \\"fieldSets\\": []
}
",
    "css": ".fragment { color: inherit; }
",
    "html": "<div>Fragment</div>
",
    "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
    "name": "Sample fragment name",
    "previewFileEntryId": 0,
    "status": 0,
    "type": 0,
  },
]
`;

exports[`import-generator/import updates collections if they already exist: addFragmentEntry 5`] = `
Array [
  "1234",
  "sample-collection",
  "sample-fragment",
  Object {
    "configuration": "",
    "css": ".fragment { color: inherit; }
",
    "html": "<div>Fragment</div>
",
    "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
    "name": "Sample fragment name",
    "previewFileEntryId": 0,
    "status": 0,
    "type": 0,
  },
]
`;

exports[`import-generator/import updates collections if they already exist: updateFragmentCollection 1`] = `
Array [
  "sample-collection",
  "Sample collection name",
  "Sample collection description",
]
`;

exports[`import-generator/import updates collections if they already exist: updateFragmentCollection 2`] = `
Array [
  "sample-collection",
  "Sample collection name",
  "Sample collection description",
]
`;

exports[`import-generator/import updates collections if they already exist: updateFragmentCollection 3`] = `
Array [
  "sample-collection",
  "Sample collection name",
  "Sample collection description",
]
`;

exports[`import-generator/import updates fragments if they already exist: updateFragmentCollection 1`] = `
Array [
  "sample-collection",
  "Sample collection name",
  "Sample collection description",
]
`;

exports[`import-generator/import updates fragments if they already exist: updateFragmentCollection 2`] = `
Array [
  "sample-collection",
  "Sample collection name",
  "Sample collection description",
]
`;

exports[`import-generator/import updates fragments if they already exist: updateFragmentCollection 3`] = `
Array [
  "sample-collection",
  "Sample collection name",
  "Sample collection description",
]
`;
